### DISCUSSION TITLE
treemap from C# with custom text information

### DISCUSSION THREAD

**Discussion by t1-nva** [plotly/Plotly.NET#466] [Q&A]
I have very little experience with F# and C# interaction and also the underlying Plotly concepts so please excuse my obvious ignorance here, but I have tried various things and still can't get it to work  :-)

I've successfully generated a treemap from an ELF symbol tree and I want to display custom information in the nodes. My current approach is to use TextInfo, but I can't figure out how to do something similar to 'label+value` in C#

I.e. is there a way to display both Label and Value or use some other mechanism to achieve the same/similar result?

``` C#
var textInfo = Plotly.NET.StyleParam.TextInfo.Value;

	var treemap = Plotly.NET.CSharp.Chart.Treemap<string, string, int, string, string>(
	labels,
	parents,
	Values: values, ShowSectionColorScale: true,
	SectionColorScale: StyleParam.Colorscale.Viridis,
	BranchValues: StyleParam.BranchValues.Total,
	SectionOutlineColor: Color.fromString("Black"),
	Tiling: TreemapTiling.init( Packing: StyleParam.TreemapTilingPacking.Squarify),
	ShowLegend: true,
	TextPosition: TextPosition.MiddleCenter,
	TextInfo: textInfo
	).WithSize( 1280, 1024 );
```

**Comment by kMutagene** [comment]
hey @t1-nva 

`StyleParam.TextInfo` has the `toConcatString` function, which you should be able to use from C# :

```csharp
using Plotly.NET;

StyleParam.TextInfo.toConcatString(
    new StyleParam.TextInfo [] {
        StyleParam.TextInfo.Label,
        StyleParam.TextInfo.Value
    }
).ToString()
```

returns

```csharp
"label+value"
```

however, we moved away from these combinatory methods in StyleParam and you seem to have found one were we simply forgot to clean the API and just expose all combination as UnionCases, so ill make sure to track this so that there will be `StyleParam.TextInfo.LabelValue` etc. in the future ;)
